x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/prebuilt_rules_install_update_workflows.mdx

```diff
@@ -0,0 +1,180 @@
+# Rule Immutability / Customization
+
+## Test Plan for 2nd Milestone of Customizing prebuilt detection rules
+
+### Useful information
+
+Ticket: [https://github.com/elastic/security-team/issues/1974](https://github.com/elastic/security-team/issues/1974)
+
+**Assumptions**
+- The current test plan is only for Milestone 2 of the Rule Immutability/Customization feature to be released in 8.9. It does not pretend to cover any scenario for past or future milestones. Scenarios and flows are sensitive to change in future Milestones.
+- Below scenarios only apply to prebuilt rules.
+- Most of our users are on the 7.17.x version, that’s why the 8.x version is specified on scenarios, because this TestPlan is considering a minimum version of 8.x.
+- The rule Customization feature should be available to users on the Basic license and higher.
+
+
+### Scenarios
+
+### Notifications
+
+#### **Scenario: No callout messages are displayed when user does not have prebuilt rules installed**
```

Comment:
For the notifications scenarios, we will have to be clear about the terminology: we need to separate "have prebuilt rules installed" from "have prebuilt rules package installed".

For example, in this scenario, I think you meant "when user does not have prebuilt rules installed", because if the user does not have prebuilt rules installed, we **should** actually show a callout (or the number of available rules to install as a badge in the Add Elastic Rules).

This scenario that you are describing here can actually not happen because when the user navigates to the Rules Management page (or directly to the Add Elastic Rules page, as well) the prebuilt rules package is installed in the background and that makes all the rules in the package available for install. Which triggers the UI to display the callout (and the badges with the number of rules).

Reply:
Synced with @vgomez-el: only case when no prebuilt rules are installed and we don't show callouts or badge with number of avaialble ruels is if the package installation fails. So we are deleting this scenario and creating a new scenario where the package isntallation fails and no callouts are shown (in the section below Installation/Update failures).

Reply:
@jpdjere @vgomez-el I don't think this thinking is accurate. We show or hide the callous and CTAs based on the result of the `/status` endpoint. This endpoint doesn't directly depend on the package, whether its installation failed or succeeded. The endpoint depends on the installed rules (`alert` saved objects) and prebuilt rule assets (`security-rule` saved objects).

There can be different edge cases:

- Let's say the package was installed in the past and the user installed _all_ the prebuilt rules from it. Now, if a new package gets released and its installation fails for some reason, we'd still _not show_ any callouts because from the `/status` endpoint's perspective, all the rules are installed and up to date.
- Let's say the package was installed in the past and the user installed _some of_ the prebuilt rules from it. Now, if a new package gets released and its installation fails for some reason, we'd still _show_ a callout with the rest of the rules available for installation.
- Etc - you can think about other permutations of user actions between two points in time: when the package was available and was installed, and when the package installation started to fail.

Conclusion: the callouts and CTAs don't depend on the latest attempt to install the package.

Let's think if we should cover this with any scenarios.

Reply:
After going through the test plan with @jpdjere on Monday, we decided that we don't really need scenarios for error handling when package installation fails. This is trivial behavior (it fails - we show a toast) and it's decoupled from the prebuilt rule workflows.

Instead, we added this one:

#### **Scenario: User is NOT notified when no prebuilt rules are installed and there are no prebuilt rules assets**

**Automation**: 1 e2e test with mock rules + 1 integration test with mock rules for the /status endpoint.

```Gherkin
Given no prebuilt rules are installed in Kibana
And no prebuilt rule assets exist
When user opens the Rule Management page
Then user should NOT see a CTA to install prebuilt rules
And user should NOT see a number of rules available to install
And user should NOT see a CTA to upgrade prebuilt rules
And user should NOT see a number of rules available to upgrade
And user should NOT see the Rule Updates table
```